common/install_sage.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# !!!NOTE!!! This script is intended to be run with root privileges
+# It will run as the 'sage' user when the time is right.
+SAGE_SRC_TARGET=${1%/}
+BRANCH=$2
+
+if [ -z $SAGE_SRC_TARGET ]; then
+  >&2 echo "Must specifiy a target directory for the sage source checkout"
+  exit 1
+fi
+
+if [ -z $BRANCH ]; then
+  >&2 echo "Must specify a branch to build"
+  exit 1
+fi
+
+N_CORES=$(cat /proc/cpuinfo | grep processor | wc -l)
+
+export SAGE_FAT_BINARY="yes"
+export MAKE="make -j${N_CORES}"
+cd "$SAGE_SRC_TARGET"
+git clone --depth 1 --branch ${BRANCH} https://github.com/sagemath/sage.git
+chown -R sage:sage sage
+cd sage
+
+# Sage can't be built as root, for reasons...
+sudo -H -E -u sage make || exit 1
```

Comment:
Note--I've reworked this script so that it is run as root (*before* the `USER sage` command in the Dockerfile).  This gives it freedom to perform operations that should be done as root, like creating links in `/usr/bin/`.  However it runs the sage build as the sage user.

This means there's actually no reason for the sage user to be in the sudoers list in the first place, and we could remove that entirely (though this PR doesn't do that yet).